projects/packages/waf/src/class-waf-runtime.php

```diff
@@ -240,15 +240,17 @@ public function match_targets( $transforms, $targets, $match_operator, $match_va
 	public function block( $action, $rule_id, $reason, $status_code = 403 ) {
 		if ( ! $reason ) {
 			$reason = "rule $rule_id";
+		} else {
+			$reason = $this->sanitize_output( $reason );
 		}
 
 		$this->write_blocklog( $rule_id, $reason );
 		error_log( "Jetpack WAF Blocked Request\t$action\t$rule_id\t$status_code\t$reason" );
-		header( "X-JetpackWAF-Blocked: $status_code $reason" );
+		header( "X-JetpackWAF-Blocked: $status_code - rule $rule_id" );
 		if ( defined( 'JETPACK_WAF_MODE' ) && 'normal' === JETPACK_WAF_MODE ) {
 			$protocol = isset( $_SERVER['SERVER_PROTOCOL'] ) ? wp_unslash( $_SERVER['SERVER_PROTOCOL'] ) : 'HTTP';
 			header( $protocol . ' 403 Forbidden', true, $status_code );
-			die( "rule $rule_id" );
+			die( "rule $rule_id - reason $reason" );
```

Comment:
Not major but if there is no reason available, I believe we'll have a redundant output here of `rule $rule_id - reason rule $rule_id`.

Reply:
@nateweller Which part is redundant? This should just be adding the reason to the die which was information we weren't displaying previously. The rule ID is already printed out to console (so technically that shows up twice), but I didn't add that. Plus, the die is the only thing that will be seen when it is executed on a site as opposed to running from a terminal curl I believe?

Reply:
Just the minor fact that the `die()` string output will repeat itself by including the "rule $rule_id" string as both the prefix and the reason:

"rule ABC123 - reason rule ABC123"

Due to `$reason` being set as "rule ABC123" when no more detailed reason is available.

Reply:
Ah - right, though that's only if there isn't a reason. There should however always be a reason provided. Normal expectation would be "rule xx - reason blah blah blah" which is just a little more helpful for troubleshooting. The die as it exists won't be a part of our final product, so this is more for testing.

Reply:
Sounds good to me!